specs/agents/tracing-instrumentation-db.md

```diff
@@ -1,34 +1,198 @@
-## Database spans
+## Database and Datastore spans
 
-We capture spans for various types of database/data-stores operations, such as SQL queries, Elasticsearch queries, Redis commands, etc. We follow some of the same conventions defined by OpenTracing for capturing database-specific span context, including:
+We capture spans for various types of database/data-stores operations, such as SQL queries, Elasticsearch queries, Redis commands, etc. 
+Database and datastore spans **must not have child spans** (within the same transaction), e.g. they must be exit spans for the service.
 
- - `db.instance`: database instance name, e.g. "customers". For DynamoDB, this is the region.
- - `db.statement`: statement/query, e.g. "SELECT * FROM foo"
- - `db.user`: username used for database access, e.g. "readonly_user"
- - `db.type`: database type/category, which should be "sql" for SQL databases, and the lower-cased database name otherwise.
+The following fields are relevant for database and datastore spans. Where possible, agents should provide information for as many as possible of these fields. The semantics of and concrete values for these fields may vary between different technologies. See sections below for details on specific technologies.
 
-The full database statement should be stored in `db.statement`, which may be useful for debugging performance issues. We store up to 10000 Unicode characters per database statement.
+| Field | Description | Mandatory Field |
+|-------|-------------|-----------------|
+|`name`| The name of the exit database span. For SQL operations we perform a limited parsing of the statement, and extract the operation name and outer-most table involved. Other databases and storages may have different strategies for the span name (see specific databases and stores in the sections below).| :white_check_mark: |
+|`type`|For database spans, the type should be `db`.| :white_check_mark:|
+|`subtype`|For database spans, the subtype should be the database vendor name. See details below for specific databases.| :x: |
+|`action` :red_circle:|`query`|
+| <hr/> |<hr/>|<hr/>|
+|`context.db.instance`:red_circle:| Database instance name, e.g. "customers". For DynamoDB, this is the region.| :x: |
```

Comment:
FYI, this might need to be changed depending on the conclusion of this discussion https://github.com/elastic/apm/pull/418/files